### DISCUSSION TITLE
yubikey support

### DISCUSSION THREAD

**Discussion by matjazantloga** [dani-garcia/vaultwarden#5091] [General]
hi, 

we want to integrate yubikey support into self hosted vaultwarden
i'm confused with what to enter here?
this should be general setup for all possible yubikey users i suppose
we are using google auth at the moment but would like to add yubikey support

![image](https://github.com/user-attachments/assets/e8c60320-31e3-419c-b060-064769e148f8)

thanks Matjaz

**Comment by BlackDex** [comment]
See
https://github.com/dani-garcia/vaultwarden/blob/65629a99f07ec399cd574a4f20b6e73c2ba8f2a3/.env.template#L438...L444

**Reply by matjazantloga** [reply]
thanks, i guess i also need to setup otp server ? hope not ... :)

**Reply by BlackDex** [reply]
If you follow the stuff in the comments like going to https://upgrade.yubico.com/getapikey/ it should work just fine.

**Reply by matjazantloga** [reply]
Yes, i did that , also got yubikey now confirmed as one of two steps logins
![image](https://github.com/user-attachments/assets/009fcaf2-cb53-4b4f-a252-5835560547f3)
but with client it doesnt work, no matter which one i choose.
trying to figure it out.

**Reply by kpfleming** [reply]
Which clients have you tried? Some of them don't support Yubikey unless it is NFC-capable (the mobile clients, I believe).

**Comment by matjazantloga** [comment]
bitwarden desktop client for windows
bitwarden edge plugin
bitwarden ios client
yubikey i have is nfc-capable


br Matjaz

________________________________

**Comment by matjazantloga** [comment]
this is from win edge browser plugin, when i try to use yubikey
[cid:49ee2fff-28f8-4787-8360-5064480884f1]



same with win desktop client
[cid:fdf68230-c076-47a5-ab4d-de484b340e98]

haven't tested iphone yet...
i'm sure i'm doing sth wrong, just don't know what


________________________________

**Comment by BlackDex** [comment]
I would suggest to check the Bitwarden Documentation: https://bitwarden.com/help/setup-two-step-login/

**Comment by matjazantloga** [comment]
sorry for wasting your time... it's working, just tried in inprivate edge ...
i simply forgot i was already "trusted"on machine and there was not need for second auth because of that.

Best regards, s prijaznimi pozdravi

Matjaž Antloga, Ortoscale d.o.o.

________________________________
